uvtools/dspec.py

```diff
@@ -60,11 +67,331 @@ def calc_width(filter_size, real_delta, nsamples):
+                        specify whether filtering will be performed in 2d or 1d.
+                        If filter is 1d, it will be applied across the -1 axis.
+                    fitting_options: dict
+                        dictionary with options for fitting techniques.
+                        if filter2d is true, this should be a 2-tuple or 2-list
+                        of dictionaries. The dictionary for each dimension must
+                        specify the following for each fitting method.
+                            * 'dft':
+                                'fundamental_period': float or 2-tuple
+                                    The fundamental_period of dft modes to fit. This is the
+                                    Fourier resoltion of fitted fourier modes equal to
+                                    1/FP where FP is the fundamental period. For a standard
+                                    delay DFT FP = B where B is the visibility bandwidth
+                                    FP also sets the number of
+                                    modes fit within each window in 'filter_half_widths' will
+                                    equal fw / fundamental_period where fw is the filter width.
+                                    if filter2d, must provide a 2-tuple with fundamental_period
+                                    of each dimension.
+                            * 'dayenu':
+                                No parameters necessary if you are only doing 'dayenu'.
+                                For 'dayenu_dpss', 'dayenu_dft', 'dayenu_clean' see below
+                                and use the appropriate fitting options for each method.
+                            * 'dpss':
+                                'eigenval_cutoff': array-like
+                                    list of sinc_matrix eigenvalue cutoffs to use for included dpss modes.
+                                'nterms': array-like
+                                    list of integers specifying the order of the dpss sequence to use in each
+                                    filter window.
+                                'edge_supression': array-like
+                                    specifies the degree of supression that must occur to tones at the filter edges
+                                    to calculate the number of DPSS terms to fit in each sub-window.
+                                'avg_suppression': list of floats, optional
+                                    specifies the average degree of suppression of tones inside of the filter edges
+                                    to calculate the number of DPSS terms. Similar to edge_supression but instead checks
+                                    the suppression of a since vector with equal contributions from all tones inside of the
+                                    filter width instead of a single tone.
+                            *'clean':
```

Comment:
can you add a comment that defaults can be accessed as dspec.CLEAN_DEFAULTS

Reply:
I now mention this.